### DISCUSSION TITLE
A new Swift PRC library

### DISCUSSION THREAD

**Discussion by ileitch** [MobileNativeFoundation/discussions#214] [General]
Hey all,

I just wanted to let you all know that Reddit has open-sourced a new library for doing RPC with Codable, called [CodableRPC](https://github.com/reddit/CodableRPC). The library itself isn't all that exciting, however, I think some of you might be interested in [how we're using it](https://www.reddit.com/r/RedditEng/comments/1bkxjsr/introducing_codablerpc_an_ios_ui_testing_power/) at Reddit. I know there are some folks here maintaining large iOS UI test suites that might find the RPC approach useful.

**Comment by rabc** [comment]
Curious on how you use it in the UI tests. Does that to call specific methods instead of performing a tap?

**Reply by ileitch** [reply]
We don't use it for tapping individual buttons exactly, but you can use it to programmatically skip specific screens or flows, e.g. logging in.
